device-connectors/src/testflinger_device_connectors/devices/muxpi/muxpi.py

```diff
@@ -434,22 +437,36 @@ def check_test_image_booted(self):
         test_password = self.job_data.get("test_data", {}).get(
             "test_password", "ubuntu"
         )
+        boot_check_cmd_str = self.job_data.get("provision_data", {}).get(
+            "boot_check_cmd", None
+        )
+        env = os.environ.copy()
+        if boot_check_cmd_str is not None:
+            env.update({
+                'TEST_USERNAME': test_username,
+                'TEST_PASSWORD': test_password,
+                'DEVICE_IP': self.config["device_ip"],
+            })
+            logger.info("Using custom boot check script: %r", boot_check_cmd_str)
+            boot_check_cmd = shlex.split(boot_check_cmd_str)
+        else:
+            boot_check_cmd = [
+                "sshpass",
+                "-p",
+                test_password,
+                "ssh-copy-id",
+                "-o",
+                "StrictHostKeyChecking=no",
+                "-o",
+                "UserKnownHostsFile=/dev/null",
+                "{}@{}".format(test_username, self.config["device_ip"]),
+            ]
```

Comment:
What do you think about f-stringing this as a DEFAULT_BOOT_CHECK_CMD or something, and making that the default above instead of None when we get the boot_check_cmd from the job?